.github/workflows/test.yaml

```diff
@@ -0,0 +1,36 @@
+name: Build and Test Container
+
+on: push
+
+jobs:
+  build-and-test:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Pull latest Docker image for cache
+        run: docker pull rehosting/fw2tar:latest || true
+
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Build Docker image
+        id: docker_build
+        uses: docker/build-push-action@v5
+        with:
+          context: .
+          push: false
+          tags: |
+            rehosting/fw2tar:${{ github.sha }}
+            rehosting/fw2tar:latest
+          cache-from: type=registry,ref=rehosting/fw2tar:latest
+          cache-to: type=inline
+
+      - name: Run Tests
+        run: ./tests/end_to_end.sh
```

Comment:
If you care, you could use a custom container name here. I don't think it matters, but Luke tells me docker sometimes does weird things when there's an org name and a latest tag. If you changed it you'd set a `--image` arg specifying the name in the `fw2tar` command in end_to_end.sh.

Reply:
I don't see what issue it could cause, and I mostly just was going for 1:1 behavior with the build-and-release as I'd like what we release to be what we test. If this causes issues down the road though this is good to know so I can change it